ocs_ci/ocs/cluster.py

```diff
@@ -912,6 +912,69 @@ def get_osd_utilization():
     return osd_filled
 
 
+def get_ceph_df_detail():
+    """
+    Get ceph osd df detail
+    :return: dict returned from cmd
+    """
+    ceph_cmd="ceph df detail"
+    ct_pod = pod.get_ceph_tools_pod()
+    output = ct_pod.exec_ceph_cmd(ceph_cmd=ceph_cmd)
+    return output
+
+
+def validate_replica_data(pool_name, replica):
+    """
+    Check if data is replica 2 or 3
+    :param int replica: size of the replica(2,3)
+    :param string pool_name: name of the pool to check replica
+    :return: true if replica is verified, false if not
+    """
+    output = get_ceph_df_detail()
+    for pool in output.get('pools'):
+        if pool.get('name') == pool_name:
+            logger.info(f"{pool_name}")
+            stored = pool.get('stats').get('stored')
+            byte_used = pool.get('stats').get('bytes_used')
+            compress_bytes_used = pool.get('stats').get('compress_bytes_used')
+            compress_under_bytes = pool.get('stats').get('compress_under_bytes')
+            byte_used = byte_used + compress_under_bytes - compress_bytes_used
+            store_ratio = byte_used / stored
+            if (replica+0.2) > store_ratio > (replica-0.2):
+                logger.info(f"pool {pool_name} meet rep {replica} size")
+                return True
+            else:
+                logger.info(f"pool {pool_name} meet do not meet rep {replica} size"
+                            f"Store ratio is {store_ratio}")
+                return False
+
+
+def validate_compression(pool_name):
+    """
+    Check if data was compressed
+    :param int replica: size of the replica(2,3)
+    :param string pool_name: name of the pool to check replica
+    :return: True if compression ratio is right, false if compression ratio is wrong
+    """
+    output = get_ceph_df_detail()
+    for pool in output.get('pools'):
+        if pool.get('name') == pool_name:
+            logger.info(f"{pool_name}")
+            byte_used = pool.get('stats').get('bytes_used')
+            compress_bytes_used = pool.get('stats').get('compress_bytes_used')
+            compress_under_bytes = pool.get('stats').get('compress_under_bytes')
+            all_byte_used = byte_used + compress_under_bytes - compress_bytes_used
+            assert compress_bytes_used != 0 or compress_under_bytes !=0, f"pool {pool_name} have stats with no compression"
+            compression_ratio = byte_used / all_byte_used
+            logger.info(f"this is the comp_ratio {compression_ratio}")
+            if 0.6 < compression_ratio:
+                f"Compression ratio {compression_ratio} is larger than 0.6"
+                return True
+            else:
+                f"Compression ratio {compression_ratio} is smaller than 0.6"
```

Comment:
I guess that these lines are missing `logger.info()` (or `debug`) calls.

Reply:
added